OpenRA.Mods.Common/FileExtensions.cs

```diff
@@ -0,0 +1,76 @@
+#region Copyright & License Information
+
+/*
+ * Copyright 2007-2022 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+
+#endregion
+
+using System;
+using System.IO;
+using System.Linq;
+
+namespace OpenRA.Mods.Common
+{
+	public static class FileInsensitive
```

Comment:
I'm surprised that this needs to be implemented from scratch. Is there really not an existing solution?

Reply:
I was also extreme surprised, when i found out that .net does not ship this by default.

Reply:
It would be good to have a comment here explaining what you mention in the PR description